crates/mun_syntax/src/ast/generated.rs

```diff
@@ -100,6 +100,79 @@ impl ArrayType {
     }
 }
 
+// AssociatedItem
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct AssociatedItem {
+    pub(crate) syntax: SyntaxNode,
+}
+
+impl AstNode for AssociatedItem {
+    fn can_cast(kind: SyntaxKind) -> bool {
+        matches!(kind, FUNCTION_DEF)
+    }
+    fn cast(syntax: SyntaxNode) -> Option<Self> {
+        if Self::can_cast(syntax.kind()) {
+            Some(AssociatedItem { syntax })
+        } else {
+            None
+        }
+    }
+    fn syntax(&self) -> &SyntaxNode {
+        &self.syntax
+    }
+}
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum AssociatedItemKind {
+    FunctionDef(FunctionDef),
+}
+impl From<FunctionDef> for AssociatedItem {
+    fn from(n: FunctionDef) -> AssociatedItem {
+        AssociatedItem { syntax: n.syntax }
+    }
+}
+
+impl AssociatedItem {
+    pub fn kind(&self) -> AssociatedItemKind {
+        match self.syntax.kind() {
+            FUNCTION_DEF => {
+                AssociatedItemKind::FunctionDef(FunctionDef::cast(self.syntax.clone()).unwrap())
+            }
+            _ => unreachable!(),
+        }
+    }
+}
+
+impl AssociatedItem {}
```

Comment:
Any idea why this empty `impl` block is generated?

Reply:
That is due to how we generate code. We only add some functions to this impl if there is stuff to implement. There are more examples of this in the file.